### PR TITLE
CPRI-4210: User Rights and Permissions

### PR DESCRIPTION
**Pull request by RussellWaterson**

### CODE REVIEW

File: src/components/AssetPlanRequest/SourceInfoSection/__test__/SourceInfoSection.test.js [hunk 1]
```diff
@@ -25,8 +25,24 @@ import {
...
+    canEditAssetRequirement: true,
+    canDeleteAssetRequirement: true,
+    canImportAssetRequirement: true,
+    canViewAssetRequirement: true
+  },
+  displayRole: 'Admin'
+}
 const uiFactory = new UIFactory()
-uiFactory.init(taxonomies, sinon.spy(), sinon.spy(), sinon.spy(), sinon.spy())
+uiFactory.init(taxonomies, sinon.spy(), sinon.spy(), sinon.spy(), sinon.spy(), user)
```
Comment (devu) [inline src/components/AssetPlanRequest/SourceInfoSection/__test__/SourceInfoSection.test.js]: can we pass just permissions to factory? I don't think Factory needs to know your name ;)

File: src/actions/constants.js [hunk 2]
```diff
@@ -45,3 +46,14 @@ export const RESET_NOTIFICATION = 'RESET_NOTIFICATION'
 export const UPDATE_MODAL = 'UPDATE_MODAL'
 export const UPDATE_MODAL_HANDLE_OK = 'UPDATE_MODAL_HANDLE_OK'
 export const RESET_MODAL = 'RESET_MODAL'
+
+// permissions
+export const CAN_CREATE_ASSET_REQUIREMENT = 'CAN_CREATE_ASSET_REQUIREMENT'
```
Comment (propattern) [inline src/actions/constants.js]: Lets have a quick chat on how to obtain them

File: src/utils/checkUserPermissions.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  CAN_IMPORT_ASSET_REQUIREMENT,
+  CAN_PUBLISH_DISTRIBUTABLE,
+  CAN_VIEW_ASSET_REQUIREMENT,
+} from '../actions/constants'
+
+const checkUserPermissions = (permission, user) => {
+  if (user && user.permissions) {
+    switch (permission) {
+      case CAN_CREATE_ASSET_REQUIREMENT:
+        return user.permissions.canCreateAssetRequirement || false
```
Comment (propattern) [inline src/utils/checkUserPermissions.js]: You should be able to destructure all of these
const {permissions: {canX, canY, canZ}} = user

**Comment by propattern** [comment]
Looks good and I will approve it, but wait for @devu feedback as well.

**Pull request merged**